plugins/by-name/nvim-tree/deprecations.nix

```
@@ -0,0 +1,188 @@
+lib: {
+  imports = [
+    (
+      let
+        oldOptPath = [
+          "plugins"
+          "nvim-tree"
+          "view"
+          "hideRootFolder"
+        ];
+      in
+      lib.mkChangedOptionModule oldOptPath [
+        "plugins"
+        "nvim-tree"
+        "settings"
+        "renderer"
+        "root_folder_label"
+      ] (config: !lib.getAttrFromPath oldOptPath config)
```

This matches the documented example: https://github.com/NixOS/nixpkgs/blob/aa5367d94d4710035ce6c58a19ba5ecfd9866e8d/lib/modules.nix#L1668

However IMO it's simpler to hard-code it, without using `getAttrFromPath`:

```suggestion
      ] (config: !config.plugins.nvim-tree.view.hideRootFolder)
```

I'm guessing the example uses that because other rename/deprecation functions take a generic option-path and can't hard-code the attrpath?

---

Although, I guess you'd need to declare `oldOptPath` anyway, for the "from" argument. So maybe my suggestion doesn't make sense after all.